content/blog/2018-10-23-react-v-16-6.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "React v16.6.0: lazy, memo and contextType"
+author: [sebmarkbage]
+---
+
+Today we're releasing React 16.6 with a few new convenient features. An easier way to consume Context from class components, a form of PureComponent/shouldComponentUpdate for function components and a way to do code splitting using Suspense.
+
+Check out the full [changelog](#changelog) below.
+
+## [`React.memo`](/docs/reference-react.md#reactmemo)
+
+Class components can bail out from rendering when their input props are the same using [`PureComponent`](/docs/react-api.html#reactpurecomponent) or [`shouldComponentUpdate`](/docs/react-component.html#shouldcomponentupdate). Now you can do the same with function components by wrapping them in [`React.memo`](/docs/reference-react.md#reactmemo).
+
+```js
+const MyComponent = React.memo(function MyComponent(props) {
+  /* only rerenders if props change */
+});
+```
+
+## [`React.lazy`](/docs/code-splitting.md#reactlazy) - Code-Splitting with `Suspense`
+
+You may have seen [Dan's talk about React Suspense at JSConf Iceland](/blog/2018/03/01/sneak-peek-beyond-react-16.html). Now you can use the Suspense component to do [code-splitting](/docs/code-splitting.md#reactlazy) by wrapping a dynamic import in a call to `React.lazy()`.
+
+```js
+import React, {lazy, Suspense} from 'react';
+const OtherComponent = lazy(() => import('./OtherComponent'));
+const MyComponent = () => (
+  <Suspense fallback={<div>Loading...</div>}>
+    <OtherComponent />
+  </Suspense>
+);
+```
+
+The Suspense component will also allow library authors to start building data fetching with Suspense support.
+
+> Note: This feature is not yet available for server-side rendering. Suspense support will be added in a later release.
+
+## [`static contextType`](/docs/context.md#classcontexttype)
+
+In [React 16.3](/blog/2018/03/29/react-v-16-3.html) we introduced the official Context API as a replacement to the previous [Legacy Context](/docs/legacy-context.html) API.
+
+```js
+const MyContext = React.createContext();
+```
+
+We've heard feedback that adopting the new render prop API can be difficult in class components. So we've add a convenience API to [consume a context value from within a class component](/docs/context.md#classcontexttype).
+
+```js
+class MyClass extends React.Component {
+  static contextType = MyContext;
+  componentDidMount() {
+    let value = this.context;
+    /* perform a side-effect at mount using the value of MyContext */
+  }
+  componentDidUpdate() {
+    let value = this.context;
+    /* ... */
+  }
+  componentWillUnmount() {
+    let value = this.context;
+    /* ... */
+  }
+  render() {
+    let value = this.context;
+    /* render something based on the value of MyContext */
+  }
+}
+```
+
+## [`static getDerivedStateFromError()`](/docs/react-component.html#static-getderivedstatefromerror)
```

Comment:
I thought we wanted this to be less prominent? Just a bullet in the changelog.

Reply:
I didn't realize it's in the docs. I think we need to call out here that it doesn't work in server rendering but that this is the plan. We should also note this in the docs.

Reply:
Huh?